io/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+#![doc = include_str!("../README.md")]
+#![deny(missing_docs, missing_debug_implementations)]
+#![no_std]
+
+use core::ffi::c_void;
+use mc_sgx_core_sys_types::sgx_status_t;
+use mc_sgx_core_types::Error;
+use mc_sgx_util::ResultInto;
+
+/// Standard Error stream
+#[derive(Debug)]
+pub struct StdError;
+
+impl StdError {
+    /// Attempts to write the entire buffer into the hosts stderr sink.
+    ///
+    /// # Arguments
+    /// * `buffer` - The buffer to write.
+    ///
+    /// # Errors
+    /// If there is any error writing all of `buffer` to the sink. No
+    /// assumptions should be made about the amount that was written to the sink
+    /// when an error occurs.
+    pub fn write_all(&self, buffer: &[u8]) -> Result<(), Error> {
+        unsafe { ocall_stderr(buffer.as_ptr() as *const c_void, buffer.len()) }.into_result()
+    }
+}
+
+extern "C" {
+    /// The ocall to send stderr messages to
+    ///
+    /// # Arguments
+    /// * `input` - The input buffer/stream. Should be ui8/bytes
+    /// * `len` - The byte length of `input`
+    ///
+    /// # Returns
+    /// `sgx_status_t::SGX_SUCCESS` when all of input was successfully written
+    /// to the untrusted stderr sink.
+    /// An error status if not all of the data could be written to the sink. No
+    /// assumptions are made about how much data was written on error.
+    fn ocall_stderr(input: *const c_void, len: usize) -> sgx_status_t;
+}
+
+// Done out here so that `serial_test` works, since it uses "::std" in the macro
+#[cfg(test)]
+extern crate std;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use core::slice;
+    use once_cell::sync::Lazy;
+    use serial_test::serial;
+    use std::string::String;
+    use std::sync::Mutex;
+
+    static TEST_STREAM: Lazy<Mutex<String>> = Lazy::new(|| Mutex::new(String::new()));
+    static TEST_STREAM_RESULT: Lazy<Mutex<sgx_status_t>> =
+        Lazy::new(|| Mutex::new(sgx_status_t::SGX_SUCCESS));
+
+    fn reset_test_stream() {
+        let mut stream = TEST_STREAM.lock().expect("Mutex has been poisoned");
+        stream.clear();
+        let mut status = TEST_STREAM_RESULT.lock().expect("Mutex has been poisoned");
+        *status = sgx_status_t::SGX_SUCCESS;
+    }
+
+    #[no_mangle]
+    extern "C" fn ocall_stderr(input: *const c_void, len: usize) -> sgx_status_t {
+        let bytes = unsafe { slice::from_raw_parts(input as *const u8, len) };
+        let message =
+            std::str::from_utf8(bytes).expect("Expected valid UTF8 from stderr in enclave");
+        let mut stream = TEST_STREAM.lock().expect("Mutex has been poisoned");
+        stream.clear();
+        stream.push_str(message);
+        let status = TEST_STREAM_RESULT.lock().expect("Mutex has been poisoned");
+        *status
+    }
```

Comment:
This allows us to test outside of an enclave.

I'm thinking we should be able to build `mc-sgx-io` and link in the enclave version of the ocall.
Then later on link that rs into the final app enclave, so we don't need to share/consolidate edl files at the app level.